client/support/support-user/index.jsx

```diff
@@ -52,21 +57,18 @@ const SupportUser = React.createClass( {
 	}
 } );
 
-const mapStateToProps = ( state ) => {
-	return {
-		isSupportUser: state.support.isSupportUser,
-		isTransitioning: state.support.isTransitioning,
-		showDialog: state.support.showDialog,
-		errorMessage: state.support.errorMessage,
-	};
-}
+const mapStateToProps = state => ( {
+	isEnabledForUser: currentUserHasFlag( state, 'calypso_support_user' ),
+	isSupportUser: state.support.isSupportUser,
+	isTransitioning: state.support.isTransitioning,
+	showDialog: state.support.showDialog,
+	errorMessage: state.support.errorMessage,
+} );
 
-const mapDispatchToProps = ( dispatch ) => {
-	return {
-		supportUserTokenFetch: fetchToken,
-		supportUserRestore: rebootNormally,
-		supportUserToggleDialog: flowRight( dispatch, supportUserToggleDialog ),
-	};
-}
+const mapDispatchToProps = dispatch => ( {
+	supportUserTokenFetch: fetchToken,
+	supportUserRestore: rebootNormally,
+	supportUserToggleDialog: flowRight( dispatch, supportUserToggleDialog ),
+} );
```

Comment:
Connect supports the second argument as a [function or object](https://github.com/reactjs/react-redux/blob/master/docs/api.md#arguments), eg:
```javascript
{
    supportUserTokenFetch: fetchToken,
    supportUserRestore: rebootNormally,
    supportUserToggleDialog
}
```